app/src/main/java/de/tum/in/tumcampusapp/managers/LocationManager.java

```diff
@@ -335,4 +337,47 @@ private Location getNextLocation() {
         }
         return Optional.absent();
     }
+
+    /**
+     * Get Building ID accroding to the current location
+     * @return the id of current building
+     */
+    public String getBuildingIDFromCurrentLocation(List<Map<String, String>> buildingsToGps){
+        return getBuildingIDFromLocation(getCurrentLocation(), buildingsToGps);
+    }
+
+    /**
+     * Get Building ID accroding to the given location
+     * @param location the give location
+     * @return the id of current building
+     */
+    private String getBuildingIDFromLocation(Location location, List<Map<String, String>> buildingsToGps){
+        final double lat = location.getLatitude();
+        final double lng = location.getLongitude();
+        float[] results = new float[1];
+        float bestDistance = Float.MAX_VALUE;
+        String bestBuilding = "";
+
+        for (Map<String, String> building : buildingsToGps) {
+            double buildingLat = Double.parseDouble(building.get(TUMBarrierFreeRequest.KEY_BUILDING_TO_GPS_LATITUDE));
+            double buildingLng = Double.parseDouble(building.get(TUMBarrierFreeRequest.KEY_BUILDING_TO_GPS_LONGITUDE));
```

Comment:
Never do requests to the API in a loop. Then again why are you doing TWO SEPARATE REQUESTS?!? 

Please combine this into one request. Maybe you can do this whole finding of the best building ID on the server?